src/structures/member.ts

```diff
@@ -60,44 +68,35 @@ const baseMember: Partial<Member> = {
 };
 
 export async function createMemberStruct(
-  data: MemberCreatePayload,
+  data: DiscordGuildMember,
   guildId: string,
 ) {
   const {
-    joined_at: joinedAt,
-    premium_since: premiumSince,
-    user: userData,
-    roles,
-    deaf,
-    mute,
-    nick,
+    user,
+    joinedAt,
+    premiumSince,
     ...rest
-  } = data;
+  } = snakeKeysToCamelCase(data) as GuildMember;
 
-  const { mfa_enabled: mfaEnabled, premium_type: premiumType, ...user } =
-    data.user || {};
-
-  const restProps: Record<string, ReturnType<typeof createNewProp>> = {};
+  const props: Record<string, ReturnType<typeof createNewProp>> = {};
 
   for (const key of Object.keys(rest)) {
     // @ts-ignore index signature
-    restProps[key] = createNewProp(rest[key]);
+    props[key] = createNewProp(rest[key]);
   }
 
-  for (const key of Object.keys(user)) {
+  for (const key of Object.keys(user!)) {
```

Comment:
Are u certain?

Reply:
yes 
> The field `user` won't be included in the member object attached to `MESSAGE_CREATE` and `MESSAGE_UPDATE` gateway events.

Reply:
**won't** :eye:

Reply:
For the `MESSAGE_CREATE` event we do this: 
```ts
    const memberStruct = await structures.createMemberStruct(
      { ...payload.member, user: payload.author },
      guild.id,
    );
```
and for `MESSAGE_UPDATE` we don't create a member.